variables.tf

```
@@ -361,6 +361,33 @@ variable "advanced_threat_protection_enabled" {
   nullable    = false
 }
 
+variable "azure_files_authentication_directory_type" {
+  description = "Determines if azure_files_authentication should be enabled. Possible values are AD, AADS or AADKERB"
+  type        = string
+  default     = null
+  nullable    = true
+
+  validation {
+    error_message = "The active_directory object is only supported when directory_type is AD"
+    condition     = var.azure_files_authentication_directory_type == null ? true : contains(["AD", "AADS", "AADKERB"], var.azure_files_authentication_directory_type)
+  }
+}
+
+variable "azure_files_authentication_active_directory" {
+  description = "Parameters needed for directory type AD"
```

Struggling to find a good description here:

```suggestion
  description = "The Active Directory configuration for Azure Files identity-based authentication. Required if value of azure_files_authentication_directory_type is \"AD\"."
```

I would love to replace this with something better 😅